src/Pages/SunJuOh/Main/Main.js

```diff
@@ -1,9 +1,193 @@
 import React from 'react';
+import Nav from '../../../Components/Nav';
+import Feed from '../Feed/Feed';
 import './Main.scss';
 
 class Main extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      feedData: [],
+    };
+  }
+
+  toggleLike = feedId => {
+    const { feedData } = this.state;
+
+    const changeHeart = feedData.map(feed => {
+      if (feedId === feed.feedId) {
+        feed.isFeedLike = !feed.isFeedLike;
+      }
+      return feed;
+    });
+
+    this.setState({ feedData: changeHeart });
```

Comment:
- feed.isFeedLike 처럼 직접 접근해서 바꿔주기보단, 아래와 같이 스프레드 연산자로 복사를 해주는 것이 더 불변성을 지키면서 업데이트를 하는 방식일 것 같네요!
```suggestion
    const newFeedData = feedData.map((feed, index) => {
      return feedId === index ? {...feed, isFeedLike: !feed.isFeedLike} : feed
    })
    
    this.setState({ feedData: newFeedData });
```

Reply:
전개구문 표현에 익숙해져야겠네요! 수정했습니다!